runelite-client/src/main/java/net/runelite/client/plugins/wikilookup/WikiLookupPlugin.java

```diff
@@ -0,0 +1,182 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.wikilookup;
+
+import com.google.common.eventbus.Subscribe;
+import net.runelite.api.Client;
+import net.runelite.api.MenuAction;
+import net.runelite.api.MenuEntry;
+import net.runelite.api.events.MenuEntryAdded;
+import net.runelite.api.events.MenuOptionClicked;
+import net.runelite.client.input.KeyManager;
+import net.runelite.client.menus.MenuManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.util.LinkBrowser;
+import net.runelite.client.util.Text;
+import javax.inject.Inject;
+
+@PluginDescriptor(
+	name = "Wiki Lookup",
+	enabledByDefault = false
+)
+public class WikiLookupPlugin extends Plugin
+{
+	private static final String WIKI = "Wiki";
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private MenuManager menuManager;
+
+	@Inject
+	private WikiLookupInput inputListener;
+
+	@Inject
+	private KeyManager keyManager;
+
+	private boolean hotKeyPressed = false;
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		keyManager.registerKeyListener(inputListener);
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		keyManager.unregisterKeyListener(inputListener);
+	}
+
+	@Subscribe
+	public void onMenuObjectClicked(MenuOptionClicked click)
+	{
+		if (click.getMenuOption().equals(WIKI))
+		{
+			String ourLookupName = Text.removeTags(click.getMenuTarget()).replaceAll("\\(.*?\\)", "").replace(' ', '_');
+			LinkBrowser.browse("http://oldschoolrunescape.wikia.com/wiki/" + ourLookupName);
+		}
+	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (!hotKeyPressed)
+		{
+			return;
+		}
+
+		String option = Text.removeTags(event.getOption()).toLowerCase();
+
+		if (event.getType() == MenuAction.EXAMINE_ITEM_GROUND.getId()
+			|| event.getType() == MenuAction.EXAMINE_ITEM.getId()
+			|| event.getType() == MenuAction.EXAMINE_ITEM_BANK_EQ.getId()
+			|| event.getType() == MenuAction.EXAMINE_OBJECT.getId()
+			|| event.getType() == MenuAction.EXAMINE_NPC.getId()
+			|| event.getType() == MenuAction.WIDGET_DEFAULT.getId()
+			|| event.getType() == MenuAction.WIDGET_TYPE_2.getId())
+		{
+			if ((option.equals("read journal:")
+				|| option.equals("cast")
+				|| option.equals("activate"))
+				|| option.contains("play")
+				|| option.equals("examine"))
+			{
+				addBlankOption("cancel", WIKI, event.getTarget());
+			}
+			else if (option.contains("open"))
+			{
+				addBlankOption("cancel", WIKI, event.getOption().replaceFirst("Open ", "").replaceAll("Journal", "") + "Diary");
+			}
+			else if (option.contains("guide"))
+			{
+				addBlankOption("cancel", WIKI, event.getOption().replaceFirst("View ", "").replaceFirst(" guide", ""));
+			}
+		}
+	}
+
+	private void changeOption(int actionID, String newOption)
+	{
+		MenuEntry[] entries = client.getMenuEntries();
+
+		for (int i = 0; i < entries.length; i++)
```

Comment:
Minor (same applies to `changeTarget`): Since you don't rely on the index of the element, you could also do a for-each loop here.

Reply:
In the if statement we are using the index to reference that entry `entries[i].setOption(newOption);` Using a foreach would be the same efficiency as well as make the code messier.

Reply:
Definitely agree that from an efficiency perspective it doesn't really matter, but just pointing out it might improve readability here.